api/handlers.go

```diff
@@ -384,17 +378,21 @@ func (api *API) userGET(w http.ResponseWriter, req *http.Request, _ httprouter.P
 
 // userLimitsGET returns the speed limits which apply to this user.
 func (api *API) userLimitsGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	t, err := tokenFromRequest(req)
-	if err != nil {
-		api.WriteJSON(w, database.UserLimits[database.TierAnonymous])
-		return
+	// First check for an API key.
+	ak, err := apiKeyFromRequest(req)
+	if err == nil {
+		u, err := api.staticDB.UserByAPIKey(req.Context(), ak)
+		if err == nil {
```

Comment:
Why wouldn't an error here be a reason to fail?  I would think that if you can't get a user by their API key that something is wrong or at least the request is invalid.

Reply:
Makes sense. If there is a supplied API key and we cannot match it to a user we'll return `anonymous` limits.